test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1255,6 +1255,44 @@ TEST_P(ExtProcIntegrationTest, GetAndRespondImmediatelyWithBadStatus) {
   EXPECT_EQ("{\"reason\": \"Because\"}", response->body());
 }
 
+// Test the filter using an ext_proc server that responds to the request_header message
+// by sending back an immediate_response message with system header mutation.
+TEST_P(ExtProcIntegrationTest, GetAndRespondImmediatelyWithSystemHeaderMutation) {
+  proto_config_.mutable_mutation_rules()->mutable_disallow_is_error()->set_value(true);
+  proto_config_.mutable_mutation_rules()->mutable_disallow_system()->set_value(true);
+  initializeConfig();
+  HttpIntegrationTest::initialize();
+  auto response = sendDownstreamRequest(absl::nullopt);
+  processAndRespondImmediately(*grpc_upstreams_[0], true, [](ImmediateResponse& immediate) {
+    immediate.mutable_status()->set_code(envoy::type::v3::StatusCode::Unauthorized);
+    auto* hdr = immediate.mutable_headers()->add_set_headers();
+    // Mutating system header :scheme is not allowed.
+    hdr->mutable_header()->set_key(":scheme");
```

Comment:
The `:scheme` header is illegal on the response. Please use the `:status` header.

Reply:
Sure, it looks like codec has it's own check about what headers can be sent back to client. And it rejected a response with :scheme in it.
Changed to :foo, which can be testing here: https://github.com/envoyproxy/envoy/blob/f00166c4cc95b5d3aea14af40a7afe0e849cc78d/source/extensions/filters/common/mutation_rules/mutation_rules.cc#L82

Reply:
BTW, there are existing comprehensive mutation_rule test in test/extensions/filters/common/mutation_rules/mutation_rules_test.cc
google3/third_party/envoy/src/test/extensions/filters/http/ext_proc/mutation_utils_test.cc
test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc
test/extensions/filters/http/ext_proc/filter_test.cc

So, the tests we added are just to verify now the immediate response are also subject to the filter mutation rule checks.